pkg/linter/messages.yaml

```diff
@@ -4149,22 +4149,32 @@ LintCode:
       Future<void> createDir(String path) async {}
       ```
   do_not_use_environment:
-    problemMessage: "Invalid use of an environment declaration."
-    correctionMessage: "Try removing the environment declaration usage."
+    problemMessage: "Avoid using environment values like '{0}' which create hidden global state."
+    correctionMessage: "Try using 'Platform.environment' for runtime access or remove environment-dependent code."
     state:
       stable: "2.9"
     categories: [errorProne]
     hasPublishedDocs: false
     deprecatedDetails: |-
-      Using values derived from the environment at compile-time, creates
+      Using values derived from environment variables at compile-time creates
```

Comment:
This sound like those constructors derive values from the *process environment* at compile-time.
They do not. They access the *compilation environment* which is guaranteed to contain, fx,
`bool.fromEnvironment('dart.library.core')`. There is nothing wrong with using this, it's a recommended way of knowing whether a platform library is available.

So `const String.fromEnvironment('APPENGINE_RUNTIME')` will not find a process environment entry like `export APPENGINE_RUNTIME=foobar`, but it will reflect a `dart -DAPPENGINE_RUNTIME=foobar` passed to the compiler at compile time.

That's working as intended.